src/operator/fully_connected-inl.h

```diff
@@ -108,9 +110,12 @@ class FullyConnectedOp : public Operator {
           Shape2(oshape[0], oshape.ProdShape(1, oshape.ndim())), s);
     }
 
-    // Legacy approach shown here for comparison:
-    //   out = dot(data, wmat.T());
+#if (MSHADOW_USE_CBLAS == 0)
+    // Legacy approach for amalgamation build w/out cblas
+    out = dot(data, wmat.T());
```

Comment:
Question is where we draw the line. linalg is all blas/lapack. And we certainly don't want to provide non-blas versions for everything. We can do it with gemm as this is the thing that is used everywhere. 
Seems like we should scope it somehow of what we all want to support also with non-blas implementations. Any suggestions? Should we ask Steffen's team whether they can add this?